agent/lib_predis.c

```diff
@@ -632,14 +632,55 @@ NR_PHP_WRAPPER(nr_predis_aggregateconnection_getConnection) {
 }
 NR_PHP_WRAPPER_END
 
+static char* nr_predis_version(zval* app TSRMLS_DC) {
+  char* retval = NULL;
+  zval* version = NULL;
+  zend_class_entry* ce = NULL;
+
+  if (0 == nr_php_is_zval_valid_object(app)) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application object is invalid",
+                     __func__);
+    return NULL;
+  }
+
+  ce = Z_OBJCE_P(app);
+  if (NULL == ce) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application has NULL class entry",
+                     __func__);
+    return NULL;
+  }
+
+  version = nr_php_get_class_constant(ce, "VERSION");
+  if (NULL == version) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application does not have VERSION",
+                     __func__);
+    return NULL;
+  }
+
+  if (nr_php_is_zval_valid_string(version)) {
+    retval = nr_strndup(Z_STRVAL_P(version), Z_STRLEN_P(version));
+  } else {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "%s: expected VERSION be a valid string, got type %d",
+                     __func__, Z_TYPE_P(version));
+  }
+
+  nr_php_zval_free(&version);
+  return retval;
+}
+
 NR_PHP_WRAPPER(nr_predis_client_construct) {
+  char* version;
   zval* conn = NULL;
   zval* params = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
   zval* scope = nr_php_scope_get(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
 
   (void)wraprec;
 
   NR_PHP_WRAPPER_CALL;
+  version = nr_predis_version(scope TSRMLS_CC);
```

Comment:
This case is covered in the `nr_php_packages_create()` function because when NULL is passed in as the version, the function will set the version to an empty string: https://github.com/newrelic/newrelic-php-agent/blob/65ca1de83400be2c7bc6bc0dbce28b1ff56ecd1d/axiom/nr_php_packages.c#L42